assayist/processor/base.py

```diff
@@ -41,20 +41,24 @@ def main(self):
     def run(self):
         """Implement analyzer code here in your subclass."""
 
-    def read_metadata_file(self, in_file):
+    def read_metadata_file(self, in_file, in_dir=None):
         """
         Read and return the specified json metadata file or an empty dict.
 
-        :param str in_file: The path to the input file to read. Probably one of the class constants.
+        :param str in_file: The name of the input file to read. Probably one of the class constants.
+        :param str in_dir: The directory the file is in. Defaults to METADATA_DIR.
         :return: a dict or list read from the file, or an empty dict
         :rtype: {}
         :raises ValueError: if the file was not valid json content
         """
-        if os.path.isfile(in_file):
-            with open(in_file, 'r') as f:
+        if not in_dir:
+            in_dir = self.METADATA_DIR
```

Comment:
Since this is a string constant, I guess it could be a default of `in_dir` in the function call instead of `None`.